library/events/json_message_event.nim

```diff
@@ -1,16 +1,59 @@
 
 import
-    std/json
+    system,
+    std/[json,sequtils]
 import
+    stew/[byteutils,results]
+import
+    ../../waku/common/base64,
+    ../../waku/waku_core/message,
     ../../waku/waku_core/message/message,
     ./json_base_event
 
-type JsonMessage = ref object
-  # https://rfc.vac.dev/spec/36/#jsonmessage-type
-  payload: string
-  contentTopic: string
-  version: uint
-  timestamp: int64
+type
+  JsonMessage* = ref object
+    # https://rfc.vac.dev/spec/36/#jsonmessage-type
+    payload*: Base64String
+    contentTopic*: string
+    version*: uint
+    timestamp*: int64
+    ephemeral*: bool
+    meta*: Base64String
+
+func fromJsonNode*(T: type JsonMessage, jsonContent: JsonNode): JsonMessage =
+  # Visit https://rfc.vac.dev/spec/14/ for further details
+  JsonMessage(
+    payload: Base64String(jsonContent["payload"].getStr()),
+    contentTopic: jsonContent["contentTopic"].getStr(),
+    version: uint32(jsonContent["version"].getInt()),
+    timestamp: int64(jsonContent["timestamp"].getBiggestInt()),
+    ephemeral: jsonContent["ephemeral"].getBool(),
+    meta: Base64String(jsonContent["meta"].getStr()),
+  )
+
+proc toWakuMessage*(self: JsonMessage): WakuMessage =
+  let payloadRes = base64.decode(self.payload)
+  if payloadRes.isErr():
+    raise newException(ValueError, "invalid payload format: " & payloadRes.error)
```

Comment:
Nitpick, if possible I'd avoid using exceptions:)

Reply:
Replaced by a Result in https://github.com/waku-org/nwaku/pull/2435/commits/a4f7b88bacdbd30fe7a76654d4e3a732ddeb1f66